### ISSUE TITLE
Need more explanations about backends

### ISSUE THREAD

**Issue by efficks** [go-graphite/carbonapi#395]
The readme need to includes more information on the backends configuration. I need to know how it works.
If I have multiple graphite go-carbon configured that have the same metrics. Can carbonapi automatically detect a failed host and get it's data automatically from the other hosts? Can it detect if a go-carbon is down and switch it's data source? If yes, after how many time the timeout is detected?
Also, can I have two go-carbon backend with different metrics? Will carbonapi detect from which server to get the information?
Finally, is carbonapi act like a load balancer for go-carbon or it choose the first one?
Can you document what is the algorithm to choose the right backends?

Thank you for your help. I need to clarify this for my configuration.

**Comment by Civil** [comment]
I have a plan to write at least some doc before tagging 0.12.0 release.

TLDR - if your backend is a single-instance (no clustering support, e.x. prometheus or go-carbon) - use "broadcast" type. carbonapi will broadcast it's query to a backend that potentially can contain metric (there is some heuristics, like getting list of stored "tlds" + path cache for each backend) and then will try to get a full answer (wait for all backends to answer and construct something meaningful), however the logic there is very simplistic and can't be compared to any normal clustering on a backend level.

If your backend supports internal clustering - use "roundrobin" type. In this case all backends will be treated as equal, that contains the same set of data and query will be done only to one of them. It's best for databases that have internal clustering (e.x. clickhouse with clustering configured or clustered version of victoriametrics).

**Comment by Civil** [comment]
https://github.com/go-graphite/carbonapi/blob/master/doc/configuration.md
